plugins/node/opentelemetry-plugin-express/src/express.ts

```diff
@@ -218,9 +218,15 @@ export class ExpressPlugin extends BasePlugin<typeof express> {
           };
         }
         const result = original.apply(this, arguments);
-        // If the callback is never called, we need to close the span.
-        setImmediate(() => {
+        /**
+         * As this point if the callback wasn't called, that means either the
+         * layer is asynchronous (so it will call the callback later on) or that
+         * the layer directly end the http response, so we'll hook into the "finish"
+         * event to handle the later case.
+         */
+        req.res?.once('finish', () => {
```

Comment:
what if error happens, will the "finish" event cover that situation too ?

Reply:
Yeah that's the `finish` event from the underlying stream. It will be run when the `res.end()` is called.